### PR TITLE
Adding typing and some documentation to the specmanager class

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

I was on autopilot and added typing and function documentation to this class before I knew what I was doing.

### CODE REVIEW

File: src/biokbase/narrative/jobs/specmanager.py [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
+        raise_exception: bool = True,
+        allow_module_match: bool = True,
+    ):
+        """Fetch the specs for a type."""
         # if we can't find a full match for a type, try to match just the module
         if (type_id not in self.type_specs) and allow_module_match:
             type_id = type_id.split(".")[0]
         if (type_id not in self.type_specs) and raise_exception:
-            raise ValueError('Unknown type id "{}"'.format(type_id))
+            err_msg = f'Unknown type id "{type_id}"'
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/specmanager.py:45]: convert `format` to f-string; move error message construction to a variable as it gets printed out twice if you construct the message within the arguments to the error.

File: src/biokbase/narrative/jobs/specmanager.py [hunk 2]
```diff
@@ -260,29 +271,35 @@ def _repr_html_(self):
...
-                        )
+                        f"{p_def} - is a data object where the type is one of:"
+                        + json.dumps(p["allowed_types"])
                     )
+                # ??? indentation error?
                 if "allowed_values" in p:
-                    p_def = p_def + " - must be one of {}".format(
-                        json.dumps(p["allowed_values"])
+                    p_def = (
+                        f"{p_def} - must be one of {json.dumps(p['allowed_values'])}"
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/specmanager.py:302]: not sure what is going on here - why there are two checks for the same thing. Suggests that some indentation might have been added? Who knows! Anyway, no user complaints, so leaving as-is.

File: src/biokbase/narrative/jobs/specmanager.py [hunk 3]
```diff
@@ -179,21 +191,20 @@ def app_params(self, spec):
             allowed_values = list (optional),
         }
         """
-        params = []
-        for p in spec["parameters"]:
-            p_info = app_param(p)
-            params.append(p_info)
+        params = [app_param(p) for p in spec["parameters"]]
```
Comment (ialarmedalien) [inline src/biokbase/narrative/jobs/specmanager.py:194]: love me a bit of list comprehension fun

File: src/biokbase/narrative/jobs/specmanager.py (hunk 2 above)
Comment (briehl) [reply src/biokbase/narrative/jobs/specmanager.py:302]: "allowed_types" vs "allowed_values", I think.

**Review by briehl — APPROVED** [review]
Looks good. Thanks for updating some ancient code!

File: src/biokbase/narrative/jobs/specmanager.py (hunk 2 above)
Comment (ialarmedalien) [reply src/biokbase/narrative/jobs/specmanager.py:302]: Of course. Doh!

**Pull request merged**